File: poultry-pal-service/src/main/java/co/za/hlaluko/dynamics/poultry/pal/model/dto/Sale.java
package co.za.hlaluko.dynamics.poultry.pal.model.dto;

import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Sale {
  private String id;
  // For egg sales per dozen
  private Integer numberOfDozensSold;
  private Double salePricePerDozen;
  // For chicken sales
  private Integer numberOfChickensSold;
  private Double salePricePerChicken;

  private String buyerName;
  private String recordedBy;
  private PaymentStatus paymentStatus;
  private Double totalSaleAmount;
  private Date saleDate;
  private Date createdDate;
}

File: poultry_pal_plus_app/lib/models/sale.dart
class Sale {
  final String id;
  final int numberOfDozensSold;
  final double salePricePerDozen;
  final int numberOfChickensSold;
  final double salePricePerChicken;
  final String buyerName;
  final String recordedBy;
  final String paymentStatus;
  final double totalSaleAmount;
  final String saleDate;
  final String createdDate;

  Sale({
    required this.id,
    required this.numberOfDozensSold,
    required this.salePricePerDozen,
    required this.numberOfChickensSold,
    required this.salePricePerChicken,
    required this.buyerName,
    required this.recordedBy,
    required this.paymentStatus,
    required this.totalSaleAmount,
    required this.saleDate,
    required this.createdDate,
  });

  factory Sale.fromJson(Map<String, dynamic> json) {
    return Sale(
      id: json['id'],
      numberOfDozensSold: json['numberOfDozensSold'] ?? 0,
      salePricePerDozen: json['salePricePerDozen'] ?? 0,
      numberOfChickensSold: json['numberOfChickensSold'] ?? 0,
      salePricePerChicken: json['salePricePerChicken'] ?? 0,
      buyerName: json['buyerName'],
      recordedBy: json['recordedBy'],
      paymentStatus: json['paymentStatus'],
      totalSaleAmount: json['totalSaleAmount'],
      saleDate: json['saleDate'],
      createdDate: json['createdDate'],
    );
  }
}
